app/updatemanager_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package app
+
+import (
+	"context"
+	"github.com/mendersoftware/mender/dbus"
+	"github.com/mendersoftware/mender/dbus/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"sync"
+	"testing"
+	"time"
+)
+
+// TestControlMap tests the ControlMap structure, and verifies the thread-safety
+// of the data access, and writes.
+func TestControlMap(t *testing.T) {
+	cm := NewControlMap()
+	var wg sync.WaitGroup
+	wg.Add(3)
+	go func() {
+		cm.Set(&UpdateControlMap{
+			ID:       "foo",
+			Priority: 0,
+		})
+		wg.Done()
+	}()
+	go func() {
+		cm.Set(&UpdateControlMap{
+			ID:       "foobar",
+			Priority: 1,
+		})
+		wg.Done()
+	}()
+	go func() {
+		cm.Set(&UpdateControlMap{
+			ID:       "foobarbaz",
+			Priority: 3,
+		})
+		wg.Done()
+	}()
+	wg.Wait()
+	assert.Equal(t, cm.Get("foo0"), &UpdateControlMap{ID: "foo", Priority: 0}, cm.controlMap)
```

Comment:
If you really want to verify thread safety, this is kind of thing that should be repeated 10000 times. :-)